src/penn_chime/models.py

```diff
@@ -186,7 +193,7 @@ def gen_policy(self, p: Parameters) -> Sequence[Tuple[float, int]]:
         ]
 
     def run_projection(self, p: Parameters, policy: Sequence[Tuple[float, int]]):
```

Comment:
Not your code changes, but I'm having trouble tracing the relevant final model state.

self.raw and self.current_infected are the only written state here. Perhaps return (raw, current_infected) instead.

This frees run_projection from the mutating any Model state.

Callers explicitly set `self.raw, self.current_infected = self.run_projection(...)`.

Lines 72 and 73 would then clearly show progression/estimation of i_day from doubling_time where line 112 would not.

Maybe I can decypher the use on line 156 which I don't think should change the model's state. And turn get_loss into a function.

Reply:
I just cleaned things up to hopefully make the model state easier to follow. `run_projection` now only reads model state and doesn't modify it at all. It just generates a dictionary of the relevant arrays and returns it.